operator/cmd/mesh/operator_test.go

```diff
@@ -158,3 +168,57 @@ func TestOperatorInit(t *testing.T) {
 		t.Fatalf("diff: %s", diff)
 	}
 }
+
+func MockKubernetesClients(_, _ string, _ clog.Logger) (kube.ExtendedClient, client.Client, error) {
+	extendedClient = kube.MockClient{
+		Interface: fake.NewSimpleClientset(),
+	}
+	kubeClient, _ = client.New(&rest.Config{}, client.Options{})
+	return extendedClient, kubeClient, nil
+}
+
+func TestOperatorInitDryRun(t *testing.T) {
+	tests := []struct {
+		operatorNamespace string
+		watchedNamespaces string
+	}{
+		{
+			// default nss
+			operatorNamespace: "",
+			watchedNamespaces: "",
+		},
+		{
+			operatorNamespace: "test",
+			watchedNamespaces: "test1",
+		},
+		{
+			operatorNamespace: "",
+			watchedNamespaces: "test4, test5",
+		},
+	}
+
+	kubeClients = MockKubernetesClients
+
+	for _, test := range tests {
+		t.Run("", func(t *testing.T) {
+			args := []string{"operator", "init", "--dry-run"}
+			if test.operatorNamespace != "" {
+				args = append(args, "--operatorNamespace", test.operatorNamespace)
+			}
+			if test.watchedNamespaces != "" {
+				args = append(args, "--watchedNamespaces", test.watchedNamespaces)
+			}
+
+			rootCmd := GetRootCmd(args)
+			err := rootCmd.Execute()
+			assert.NoError(t, err)
+
+			actions := extendedClient.Kube().(*fake.Clientset).Actions()
+			for _, action := range actions {
+				if action.GetVerb() != "get" {
```

Comment:
list/watch is not a problem. Or does action consider these 'get'?

Reply:
Thanks. Fixed in https://github.com/istio/istio/pull/36641